mount/mount.go

```diff
@@ -22,3 +46,39 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// TempMount mounts the provided mounts to a temp dir, and pass the temp dir to f.
+// Finally we will unmount and remove the temp dir regardless of the result of f.
+func TempMount(mounts []Mount, m Mounter, f func(root string) error) error {
+	if len(mounts) == 0 {
+		return errdefs.ErrInvalidArgument
+	}
+	if m == nil {
+		m = mounter{}
+	}
+
+	root, err := ioutil.TempDir("", "containerd-TempMount")
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s", root)
+	}
+	// We change RemoveAll to Remove so that we either leak a temp dir
```

Comment:
Use remove here to protect against problems with removeall. If the mount fails, it prevents data loss if the directory is still mounted.

Please update this comment to make it clear that is why remove is used here.

Reply:
done